pkg/issuer/acme/http/service.go

```diff
@@ -71,19 +74,24 @@ func (s *Solver) getServicesForChallenge(ctx context.Context, ch *cmacme.Challen
 		selector = selector.Add(*req)
 	}
 
-	serviceList, err := s.serviceLister.Services(ch.Namespace).List(selector)
+	serviceList, err := s.serviceLister.ByNamespace(ch.Namespace).List(selector)
 	if err != nil {
 		return nil, err
 	}
 
-	var relevantServices []*corev1.Service
+	var relevantServices []*metav1.PartialObjectMetadata
 	for _, service := range serviceList {
-		if !metav1.IsControlledBy(service, ch) {
-			logf.WithRelatedResource(log, service).Info("found existing solver pod for this challenge resource, however " +
+		// TODO: can we use a metadata specific lister instead?
```

Comment:
```suggestion
		// TODO: can we use a PartialObjectMetadata lister instead of a cache.GenericLister?
```

Reply:
the original comment was a bit confusing to me.

Reply:
I remove this in a later commit as I discoverd that we cannot because that lister isn't useful for anything

Reply:
@irbekrm what do you mean? The comment is still here. Do you mean there is still another commit that you haven't pushed yet?

Reply:
Thanks, there were a couple of these comments and apparently this one was still left in. Should be removed in https://github.com/cert-manager/cert-manager/pull/5976/commits/300fe72ff0e5d6ea442481bc5ec41faab8ebbb13